presto-main/src/main/java/com/facebook/presto/sql/planner/GroupingOperationRewriter.java

```diff
@@ -59,34 +60,37 @@ public static Expression rewriteGroupingOperation(GroupingOperation expression,
         // in the group and none of them will be aggregated over. Hence, re-write the
         // GroupingOperation to a constant literal of 0.
         // See SQL:2011:4.16.2 and SQL:2011:6.9.10.
-        boolean isSimpleGroupByPresent = queryNode.getGroupBy().get().getGroupingElements().stream().anyMatch(SimpleGroupBy.class::isInstance);
+        boolean isSimpleGroupByPresent = analysis.getGroupingSets(queryNode).size() == 1;
         if (isSimpleGroupByPresent) {
             return new LongLiteral("0");
         }
         else {
             checkState(groupIdSymbol.isPresent(), "groupId symbol is missing");
 
             Map<Expression, FieldId> columnReferenceFields = analysis.getColumnReferenceFields();
-            List<Expression> columnReferences = ImmutableList.copyOf(analysis.getColumnReferences());
             List<Expression> groupingOrdinals = expression.getGroupingColumns().stream()
-                    .map(groupingColumn -> extractGroupingColumnOrdinal(groupingColumn, columnReferences, columnReferenceFields))
-                    .map(columnOrdinal -> new LongLiteral(Integer.toString(columnOrdinal)))
+                    .peek(groupingColumn -> checkState(
+                            groupingColumn instanceof Identifier || groupingColumn instanceof DereferenceExpression,
+                            "the grouping column is neither an identifier nor a dereference expression"))
+                    .map(columnReferenceFields::get)
+                    .map(FieldId::getFieldIndex)
```

Comment:
Agree with @findepi, `checkState` should be here. However, currently `GROUP BY` expressions cannot reference columns from outer scope, so query like:
```
SELECT a, (
  SELECT a+b
  FROM (VALUES 2) u(b)
  GROUP BY a, b
)
FROM (VALUES 1) t(a)
```
will fail with:
```
Query 20170509_150917_00040_ra2yx failed: Grouping field FieldId{RelationId{AliasedRelation, x002dca6e}, 0} should originate from [u.b:integer]
```
Therefore, such check state won't be tested for now